modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/HealthCheckConsensusRound.scala

```diff
@@ -68,29 +72,35 @@ class HealthCheckConsensusRound[F[_]: Spawn: Clock, A <: HealthCheckStatus, B <:
 
   def getRoundIds: F[Set[HealthCheckRoundId]] = roundIds.get
 
-  def addParallelRounds(key: HealthCheckKey)(roundIds: Set[HealthCheckRoundId]): F[Unit] = ???
+  def addParallelRounds(key: K)(roundIds: Set[HealthCheckRoundId]): F[Unit] = ???
 
-  def calculateOutcome: F[HealthCheckConsensusDecision] = {
-    def status: A = ???
+  def calculateOutcome: F[HealthCheckConsensusDecision] =
+    status.flatMap { _status =>
+      (proposals.get, peers.get).mapN { (_proposals, _peers) =>
+        def received = _proposals.view.filterKeys(_peers.contains).values.toList
 
-    (proposals.get, peers.get).mapN { (_proposals, _peers) =>
-      def received = _proposals.view.filterKeys(_peers.contains).values.toList
-
-      driver.calculateConsensusOutcome(key, status, selfId, received)
+        driver.calculateConsensusOutcome(key, _status, selfId, received)
+      }
     }
-  }
 
   def manage: F[Unit] =
     Clock[F].monotonic.flatMap { currentTime =>
       sendProposal // Note: check elapsed time and execute additional tasks
     }
 
-  def generateHistoricalData(decision: HealthCheckConsensusDecision): F[HistoricalRound] = ???
+  def generateHistoricalData(decision: HealthCheckConsensusDecision): F[HistoricalRound[K]] = ???
 
-  def ownConsensusHealthStatus: F[B] = {
-    def status: A = ???
-    Applicative[F].pure(driver.consensusHealthStatus(key, status, roundId, selfId))
-  }
+  def ownConsensusHealthStatus: F[B] =
+    status.map {
+      driver.consensusHealthStatus(key, _, roundId, selfId)
+    }
+
+  private def status =
+    ownStatus.join.flatMap {
+      case Outcome.Succeeded(fa) => fa
+      case Outcome.Errored(e)    => MonadError[F, Throwable].raiseError[A](e)
+      case Outcome.Canceled()    => MonadError[F, Throwable].raiseError[A](new Throwable("Unexpected error"))
```

Comment:
Not sure what should be the behavior here, but keep in mind that other nodes will expect to get a status from you, if you fail with an error they will keep on waiting. So maybe it should default to some value.

Reply:
Hm, I like that idea. Will do that :+1: